source/clear-linux/guides/maintenance/kernel-development.rst

```diff
@@ -0,0 +1,365 @@
+.. _kernel-development:
+
+Kernel development on Clear Linux 
+#################################
+
+The kernel is the heart of an operating system with many things relying on it
+to provide access to hardware and software functionality. 
+The Linux kernel is very customizable and extensible. 
+ 
+The |CL| kernels aim to be performant and practical. In some cases, it may be 
+necessary modify the kernel to suit your specific needs or test new kernel 
+code as a developer.
+
+In cases where features beyond those enabled by default in |CL-ATTR| are
+needed it may be necessary to:
+
+.. contents:: :local:
+   :depth: 2
+   :backlinks: top
+
+
+Request the change be included with the |CL| kernel
+===================================================
+
+If the kernel modification you need is already open source and likely to be 
+useful to others, consider submitting a request to add or enable in the
+|CL| kernel.
+
+If your change request is accepted, you do not need to maintain your own 
+modified kernel.
+
+Make enhancement requests to the |CL| distribution `on GitHub`_ .
+
+
+Customize the Linux kernel source
+=================================
+
+In some cases, it may be necessary modify the kernel to suit your specific 
+needs or test new kernel code as a developer.
+
+You can build and install a custom kernel, however you must:
+
+* Disable Secure Boot
+* Maintain any updates to the kernel going forward
+
+To customize the kernel you will need a |CL| development environment, 
+make changes to the kernel, build the kernel, and install it.
+
+.. note::
+
+      This document shows how to obtain and compile a Linux kernel source 
+      using the |CL| development tooling.
+
+      Source RPM files (SRPM) are available for all |CL| kernels, and can be used for development instead.
+      The latest source RPM files are available at: `https://download.clearlinux.org/current/source/SRPMS/`_
+
+
+Install the |CL| development tooling framework
+----------------------------------------------
+
+.. include:: autospec.rst
+   :start-after: install-tooling-after-header:
+   :end-before: install-tooling-end:
+
+
+
+Clone the Linux kernel package 
+------------------------------
+Clone the existing kernel package re-pository from |CL| as a starting point.
+
+#. Clone the Linux kernel package from |CL|.
+
+   .. code-block:: bash
+
+      cd ~/clearlinux
+      make clone_linux
+
+
+#. Navigate into the cloned package directory.
+
+   .. code-block:: bash
+
+      cd ~/clearlinux/packages/linux
+
+.. note::
```

Comment:
Maybe try to weave this Note into the introduction?  IMO, there are a lot of "Notes" in this document.

Reply:
Agreed, there too many notes. There are a lots of spots where additional information/context is needed.  This is one of the ones I think is important to be close to the decision making point.

However, I have reduced the use of notes in other areas.